src/urllib3/_async/response.py

```diff
@@ -97,6 +97,53 @@ def _get_decoder(mode):
     return DeflateDecoder()
 
 
+class ErrorCatcher(object):
+
+    def __init__(self, response):
+        self.response = response
+        self.clean_exit = False
+
+    def __enter__(self):
+        try:
+            yield
```

Comment:
I'll be able to help more tomorrow but for now I think you need a second context manager class instead of using yield here.